controllers/hivelocitymachine_controller.go

```diff
@@ -246,10 +246,12 @@ func (r *HivelocityMachineReconciler) SetupWithManager(ctx context.Context, mgr
 
 // HivelocityClusterToHivelocityMachines is a handler.ToRequestsFunc to be used to enqeue requests for reconciliation
 // of HivelocityMachines.
-func (r *HivelocityMachineReconciler) HivelocityClusterToHivelocityMachines(ctx context.Context, log logr.Logger) handler.MapFunc {
-	return func(o client.Object) []ctrl.Request {
+func (r *HivelocityMachineReconciler) HivelocityClusterToHivelocityMachines(_ context.Context) handler.MapFunc {
+	return func(ctx context.Context, o client.Object) []ctrl.Request {
 		result := []ctrl.Request{}
 
+		log := log.FromContext(ctx)
```

Comment:
two options: either name it "logger" or use another function than log.FromContext. I believe you could use the ctrl packagr well. this way you do an import shadow